### ISSUE TITLE
can't find component in subqueries

### ISSUE THREAD

**Issue by karudo** [percyhanna/rquery#27]
``` javascript
        class MyLink extends React.Component {
            render() {
                return <a href={this.props.href}>{this.props.text}</a>;
            }
        }

        class MyComponent extends React.Component {
            render() {
                return(
                    <div className="row">
                        <div className="col">
                            <MyLink href="http://google.com/" text="Google"/>
                        </div>
                    </div>
                );
            }
        }

        const component = TestUtils.renderIntoDocument(<MyComponent/>);
        const $r = $R(component);

        $r.find('MyLink').length.should.be.equal(1); // => ok
        $r.find('.col').find('MyLink').length.should.be.equal(1); // => why?
```

**Comment by percyhanna** [comment]
Hmm, I remember running into this before, but I thought I had fixed it. I'll keep you updated.

**Comment by percyhanna** [comment]
Looks like this was a combination of two bugs. I had fixed the first part of this bug on the child (`>`) selector, but not with descendant (`  `). There's also a bug when chaining`find('div').find('MyComponent')`, vs. just doing a somewhat equivalent query in one go (e.g.`find('div MyComponent')`).

Working on a fix, should be up shortly.

**Comment by percyhanna** [comment]
This should be fixed by #28. Released as [version `4.0.0`](https://github.com/percyhanna/rquery/releases/tag/v4.0.0) as it introduces breaking changes.

**Comment by karudo** [comment]
Thank you for your time!
